py3status/modules/usbguard.py

```diff
@@ -200,182 +82,94 @@ def post_config_hook(self):
+        try:
+            raw_devices = self.proxy.listDevices("(s)", "match")
+        except Exception:
+            raise Exception(STRING_USBGUARD_DBUS)
 
-    def _manipulate_devices(self, devices, device_filter):
-        new_device = []
-        format_device_list = None
-        self.device_length = len(devices)
-        for index, device in enumerate(devices, 1):
-            if device_filter == "any":
-                pass
-            elif device_filter == "None":
-                device["policy"] = None
-            elif device["policy"] != device_filter:
-                continue
-            for x in self.init["format_button"]:
-                composite = self.py3.safe_format(getattr(self, x), device)
-                device[x] = self.py3.composite_update(
-                    composite, {"index": "{}/{}".format(device["id"], x.split("_")[-1])}
+        devices = []
+        for device_id, string in raw_devices:
+            device = {"id": device_id}
+            for name, regex in self.keys:
+                value = regex.findall(string) or None
+                if value:
+                    value = value[0]
+                    value = value.encode("latin-1").decode("unicode_escape")
+                    value = value.encode("latin-1").decode("utf-8")
+                device[name] = value
+            if device["policy"] == "block":
+                devices.append(device)
+
+        return devices
+
+    def _format_device(self, devices):
+        device_info = []
+        for device in devices:
+            for btn in self.init["format_button"]:
+                composite = self.py3.safe_format(getattr(self, btn), device)
+                device[btn] = self.py3.composite_update(
+                    composite,
+                    {"index": "{}/{}".format(device["id"], btn.split("_")[-1])},
                 )
 
-            device_composite = self.py3.safe_format(self.format_device, device)
-            if index == self.device_index:
-                format_device_list = device_composite
-
-            new_device.append(device_composite)
+            device_info.append(self.py3.safe_format(self.format_device, device))
 
         format_device_separator = self.py3.safe_format(self.format_device_separator)
-        format_device = self.py3.composite_join(format_device_separator, new_device)
-
-        return format_device, format_device_list
+        format_device = self.py3.composite_join(format_device_separator, device_info)
 
-    def _notify_user(self, device):
-        format_notification_message = self.py3.safe_format(
-            self.format_notification_message, device
-        )
-        format_notification_title = self.py3.safe_format(
-            self.format_notification_title, device
-        )
-        self.py3.notify_user(
-            msg=format_notification_message,
-            title=format_notification_title,
-            icon="/usr/share/icons/hicolor/scalable/apps/usbguard-icon.svg",
```

Comment:
RIP awesome usbguard notification system on dunst. I liked this feature. :beer:

Reply:
Yeah, but both me and cyrinux disabled it, so postponing until someone actually requests it. But we know where to copy the code, if it is needed 😉